lib/config/generator.js

```diff
@@ -0,0 +1,210 @@
+            errorList = getUniqueErrorNames(errorList);
+
+            var violatedRuleCount = errorList.length;
+
+            if (!violatedRuleCount) { return this._config; }
+
+            console.log(_path + ' violates ' + violatedRuleCount + ' rule' + (violatedRuleCount > 1 ? 's' : ''));
+
+            var errorPrompts = generateRuleHandlingPrompts(errorList);
+
+            return this._getUserViolationChoices(errorPrompts)
+            .then(this._handleViolatedRules.bind(this, errorPrompts, errorList))
+            .then(function() {
+                return this._config;
+            }.bind(this));
+        }.bind(this))
+        .then(function flushConfig(config) {
+            fs.writeFileSync(process.cwd() + '/.jscsrc', JSON.stringify(this._config, null, '\t'));
+            console.log('Generated a .jscsrc configuration file in ' + process.cwd());
+        }.bind(this));
+    },
+
+    /**
+     * @private
+     * @param  {String[]} presetNames
+     * @param  {Object[]} errorsCollection
+     */
+    _showErrorCounts: function(presetNames, errorsCollection) {
+        var errorCounts = errorsCollection.map(function(error) {
+            return error.getErrorCount();
+        });
+
+        var table = getTable();
+
+        presetNames.forEach(function(name, idx) {
+            table.push([idx + 1, name, errorCounts[idx]]);
+        });
+
+        console.log(table.toString());
+    },
+
+    /**
+     * Prompts the user to choose a preset
+     *
+     * @private
+     * @param {Object} prompt
+     * @return {Promise}
+     */
+    _getUserPresetChoice: function(prompt) {
+        return this._showPrompt(prompt);
+    },
+
+    /**
+     * Prompts the user to nullify rules or fix violations themselves
+     *
+     * @private
+     * @param  {Object[]} errorPrompts
+     * @return {Promise}
+     */
+    _getUserViolationChoices: function(errorPrompts) {
+        return this._showPrompt(errorPrompts);
+    },
+
+    /** @private */
+    _showPrompt: vowNode.promisify(prompt.get.bind(prompt)),
+
+    /**
+     * @private
+     * @param  {Object[]} errorPrompts
+     * @param  {String[]} errorList
+     * @param  {Object[]} choices
+     */
+    _handleViolatedRules: function(errorPrompts, errorList, choices) {
+        errorPrompts.forEach(function(errorPrompt, idx) {
+            var associatedRuleName = errorList[idx];
+            var userChoice = choices[errorPrompt.name];
+
+            if (userChoice.toLowerCase() === 'e') {
+                this._config[associatedRuleName] = null;
+            }
+        }, this);
+    },
+
+    /**
+     * @private
+     * @param  {String} path
+     * @param  {String} presetName
+     * @return {Promise}
+     */
+    _checkAgainstPreset: function(path, presetName) {
+        var checker = getChecker();
+
+        checker.configure({preset: presetName});
+
+        return checker.checkPath(path);
+    }
+};
+
+function getChecker() {
```

Comment:
I have a suggestion. Can we use `jscs-jsdoc` plugin a little bit later and fix all docs?

Sure it would be great if these new functions will have some description to decrease well known bus factor.